### PR TITLE
Updates with stellargraph and wait for train initiate commands to complete.

### PR DESCRIPTION
**Pull request by thevindu-w**
Updates for new base image with Python3.8 and stellargraph.
Wait for train initiate Python commands to complete.

### CODE REVIEW

File: src/server/JasmineGraphInstanceService.cpp [hunk 1]
```diff
@@ -4535,14 +4540,17 @@ void JasmineGraphInstanceService::initAgg(string trainData){
...
                                 + utils.getJasmineGraphProperty("org.jasminegraph.fl.dataDir")
                                 + " " + utils.getJasmineGraphProperty("org.jasminegraph.fl.dataDir")+ " " + "4" + " 0 "
                                 + utils.getJasmineGraphProperty("org.jasminegraph.fl.num.orgs")
                                 + " " + utils.getJasmineGraphProperty("org.jasminegraph.fl.epochs") +" localhost 5000 > " 
                                 + "/home/ubuntu/software/jasminegraph/logs/agg_logs-" + Utils::getCurrentTimestamp() + ".txt";
     instance_logger.log("Executing : " + command, "info");
-    popen(command.c_str(), "r");
+    int exit_status = system(command.c_str());
+    if (exit_status == -1) {
+        instance_logger.error("fl_client.py Execution Failed");
```
Comment (miyurud) [inline src/server/JasmineGraphInstanceService.cpp:4552]:
```suggestion
        instance_logger.error("Failed to execute organization level aggregations");
```

File: src/server/JasmineGraphInstanceService.cpp [hunk 2]
```diff
@@ -4571,7 +4579,10 @@ void JasmineGraphInstanceService::initClient(string trainData){
                                 + " > /home/ubuntu/software/jasminegraph/logs/client_logs_" + partitionID + "-" + Utils::getCurrentTimestamp() + ".txt";
 
     instance_logger.log("Executing : " + command, "info");
-    popen(command.c_str(), "r");
+    int exit_status = system(command.c_str());
+    if (exit_status == -1) {
+        instance_logger.error("fl_client.py Execution Failed");
```
Comment (miyurud) [inline src/server/JasmineGraphInstanceService.cpp:4584]:
```suggestion
        instance_logger.error("Could not start python client");
```

File: src/server/JasmineGraphInstanceService.cpp (hunk 1 above)
Comment (thevindu-w) [reply src/server/JasmineGraphInstanceService.cpp:4552]: Will change it.

File: src/server/JasmineGraphInstanceService.cpp (hunk 2 above)
Comment (thevindu-w) [reply src/server/JasmineGraphInstanceService.cpp:4584]: Will change it.

File: src/server/JasmineGraphInstanceService.cpp [hunk 3]
```diff
@@ -4491,7 +4490,10 @@ void JasmineGraphInstanceService::initServer(string trainData){
                                 + " " + utils.getJasmineGraphProperty("org.jasminegraph.fl.epochs") +" localhost 5000 > " 
                                 + "/home/ubuntu/software/jasminegraph/logs/server_logs-" + Utils::getCurrentTimestamp() + ".txt";
     instance_logger.log("Executing : " + command, "info");
-    popen(command.c_str(), "r");
+    int exit_status = system(command.c_str());
+    if (exit_status == -1) {
+        instance_logger.error("fl_client.py Execution Failed");
```
Comment (miyurud) [inline src/server/JasmineGraphInstanceService.cpp:4495]:
```suggestion
        instance_logger.error("Failed executing python server for query ");
```

Comment (thevindu-w) [reply src/server/JasmineGraphInstanceService.cpp:4495]: Will change it.

File: src/server/JasmineGraphInstanceService.cpp [hunk 4]
```diff
@@ -4511,11 +4513,14 @@ void JasmineGraphInstanceService::initOrgServer(string trainData){
...
     std::string path = "cd " + utils.getJasmineGraphProperty("org.jasminegraph.fl.location") + " && ";
-    std::string command = path + "python3.11 org_server.py " + graphID+ " " + utils.getJasmineGraphProperty("org.jasminegraph.fl_clients")
+    std::string command = path + "python3.8 org_server.py " + graphID+ " " + utils.getJasmineGraphProperty("org.jasminegraph.fl_clients")
                                     + " " + utils.getJasmineGraphProperty("org.jasminegraph.fl.epochs")
                                     +" localhost 5050 > /home/ubuntu/software/jasminegraph/logs/org_server_logs-" + Utils::getCurrentTimestamp() + ".txt";
     instance_logger.log("Executing : " + command, "info");
-    popen(command.c_str(), "r");
+    int exit_status = system(command.c_str());
+    if (exit_status == -1) {
+        instance_logger.error("fl_client.py Execution Failed");
```
Comment (miyurud) [inline src/server/JasmineGraphInstanceService.cpp:4522]:
```suggestion
        instance_logger.error("Failed executing python server for multi-organization query");
```

Comment (thevindu-w) [reply src/server/JasmineGraphInstanceService.cpp:4522]: Will change it.

**Review by miyurud — CHANGES_REQUESTED** [review]
I have put few comments. Please address those.

**Review by miyurud — APPROVED** [review]

File: src/server/JasmineGraphInstanceService.cpp (hunk 4 above)
Comment (miyurud) [reply src/server/JasmineGraphInstanceService.cpp]: Ok

File: src/server/JasmineGraphInstanceService.cpp (hunk 3 above)
Comment (miyurud) [reply src/server/JasmineGraphInstanceService.cpp]: Ok

File: src/server/JasmineGraphInstanceService.cpp (hunk 2 above)
Comment (miyurud) [reply src/server/JasmineGraphInstanceService.cpp]: Ok

File: src/server/JasmineGraphInstanceService.cpp (hunk 1 above)
Comment (miyurud) [reply src/server/JasmineGraphInstanceService.cpp]: Ok

**Pull request merged**